test/csuite/timestamp_abort/main.c

```diff
@@ -1762,11 +1762,26 @@ main(int argc, char *argv[])
              * from the time we notice that the file has been created. That allows the test to run
              * correctly on really slow machines.
              */
-            while (!testutil_exists(NULL, ckpt_file))
-                testutil_sleep_wait(1, pid);
-            sleep(timeout);
+            wait_time = 0;
             sa.sa_handler = SIG_DFL;
             testutil_assert_errno(sigaction(SIGCHLD, &sa, NULL) == 0);
+            while (!testutil_exists(NULL, ckpt_file)) {
```

Comment:
The code additions must be done before the `sigaction` above. Otherwise we wouldn't properly handle child failures. The `sigaction` should be called just before the child is killed; we want the custom handler during normal child runtime, so that we can intercept and properly handle any failures within the child.

If you need the default `SIGCHLD` handler before you abort the child process, you should reset it just before you do the `kill`.

Reply:
I put it back.Thanks.